File: _site_local/mod_10_paradigmas_e_padroes_de_projeto/exemplos/procedural/05-estruturas-repetitivas/repeticoes.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        // Criar scanner para ler a entrada do usuário
        Scanner scanner = new Scanner(System.in);

        // Solicitar o número de repetições
        System.out.print("Quantas repetições? ");
        int numRepeticoes = scanner.nextInt();

        // Inicializar variáveis
        String text = "";  // Inicializa o texto como uma string vazia
        int i = 0;         // Inicializa i com 0

        // Loop do-while
        do {
            // Adicionar ao texto
            text += "The number is " + i + "\n";  // Adiciona a frase com o número atual
            i++;
        } while (i < numRepeticoes);

        // Exibir o resultado
        System.out.println(text);

        // Fechar o scanner
        scanner.close();
    }
}

File: _site_local/mod_10_paradigmas_e_padroes_de_projeto/exemplos/procedural/05-estruturas-repetitivas/repeticoes.py
# Solicitar o número de repetições
num_repeticoes = int(input("Quantas repetições? "))

# Inicializar variáveis
text = ""  # Inicializa o texto como uma string vazia
i = 0       # Inicializa i com 0

# Loop do-while em Python (usando while True e uma condição de quebra)
while True:
    # Adicionar ao texto
    text += f"The number is {i}\n"  # Adiciona a frase com o número atual
    i += 1
    if i >= num_repeticoes:
        break

# Agora, você pode usar ou imprimir a variável text
print(text)
